index.js

```diff
@@ -256,13 +257,19 @@ class LedgerBridgeKeyring extends EventEmitter {
   }
 
   _signTransaction (address, tx, toAddress, handleSigning) {
+
     return new Promise((resolve, reject) => {
       this.unlockAccountByAddress(address)
         .then((hdPath) => {
+          const rawTx = tx.raw()
+          const rawTxVRSRemoved = rawTx.slice(0, rawTx.length - 3)
+          const serializedUntypedTx = ethUtil.rlp.encode(rawTxVRSRemoved)
```

Comment:
Why does `tx.raw()` return an invalid unsigned transaction?  `tx.raw()` IMO should always return a valid transaction (either legacy, 155, 2930, or 1559).  It should not ever return an invalid transaction which it appears to be doing.  If it always returned a valid transaction then you wouldn't have to have this awkward code that strips it here.

Reply:
@danjm can you try using `tx.getMessageToSign(false)` which will not include the v,r,s values

Reply:
@MicahZoltu seems reasonable, if the tx is unsigned we can exclude v,r,s in raw()

Reply:
Just make sure that you do include all 9 fields in the case of 155 transactions.  6 fields for legacy, 8 fields for 2930, and 9 fields for 1559.

Reply:
Thanks @ryanio I just discovered `getMessageToSign` as well. I believe it will work.

Thanks @MicahZoltu. Fortunately, ethereumjs-tx's `getMessageToSign` handles the inclusion of proper fields for 155 transactions, as well as transactions of the other types, so we should be good on that front

Reply:
// will cc me in here. 😄

Reply:
fyi, in prod on metamask v10.0.x today we are doing this: https://github.com/MetaMask/eth-ledger-bridge-keyring/blob/main/index.js#L242-L253

So ledger signing will only work for 155 transactions. Following above advice I will now update this PR so that it works for all types

Reply:
Ok, thanks for letting me know, glad that this can be simplified and improved now.

Let us know if there are remaining unclarities or if you stumble upon additional things.

Reply:
Thanks everyone. The latest two commits fix the issue in a way that takes advantage of proper use of the etherumjs-tx api and the `getMessageToSign` method.